### PR TITLE
[HLS-351] Release prep

### PR DESCRIPTION
**Pull request by karenfeng**
## What changes are proposed in this pull request?

- Make sure tests pass with Java (not Kryo) serialization
- Move README from md to rst to play nice with pip descriptions
- Prune (mostly GATK) dependencies to reduce package resolution time
- Set up pipeline to release maven artifacts to Bintray
- Set up Python wheel creation (note that `typeguard` exists in PyPi but not TestPyPi)

### Instructions

- To create a maven artifact and...
  + Publish to [Bintray](https://bintray.com/projectglow/glow)
     * Prereq: get Bintray creds (via Lastpass)
     * Run `sbt release` from glow
  + Promote to [Maven Central](https://mvnrepository.com/artifact/io.projectglow/glow)
     * Sync from [Maven Central tab in Bintray](https://bintray.com/projectglow/glow/glow#central)
- To create a wheel and...
  + Publish to [TestPyPi](https://test.pypi.org/project/glow.py/)
     * Prereq: update your conda environment so that you can use `setuptools` and `twine`
     * Run `python setup.py bdist_wheel` from glow/python
     * Run `twine upload --repository-url https://test.pypi.org/legacy/ dist/*`
  + Promote to [PyPi](https://pypi.org/project/glow.py/)
     * Run `twine upload dist/*`

## How is this patch tested?
- [ ] Unit tests
- [ ] Integration tests
- [x] Manual tests

A DBR cluster was set up with the maven artifacts and wheel, and basic functionality was tested.
- [Scala notebook](https://shard-qa.staging.cloud.databricks.com/#notebook/1432371)
- [Python notebook](https://shard-qa.staging.cloud.databricks.com/#notebook/1432206)

### CODE REVIEW

File: version.sbt [hunk 1]
```diff
@@ -0,0 +1 @@
+version in ThisBuild := "0.1.1-SNAPSHOT"
```
Comment (henrydavidge) [inline version.sbt]: Why have the version in a separate file? Also the version should be 0.1.0, as it is in build.sbt.

File: python/setup.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+setup(
+    name='glowpy',
+    version='0.1.0',
+    packages=['glow'],
+    install_requires=[
+        'typeguard==2.5.0',
+    ],
+    author='Glow Project',
+    description='Glow: Genomics on Apache Spark',
```
Comment (henrydavidge) [inline python/setup.py]: Should use same messaging as website: "An open-source toolkit for large-scale genomic analysis"

File: python/setup.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from setuptools import setup
+
+setup(
+    name='glowpy',
+    version='0.1.0',
+    packages=['glow'],
+    install_requires=[
+        'typeguard==2.5.0',
+    ],
+    author='Glow Project',
```
Comment (henrydavidge) [inline python/setup.py]: "The Glow Authors" (same as copyright in license header)

File: build.sbt [hunk 4]
```diff
@@ -127,33 +158,37 @@ lazy val python =
...
-      }
+      },
+      publish / skip := true
     )
 
-// Uncomment the following for publishing to Sonatype.
-// See https://www.scala-sbt.org/1.x/docs/Using-Sonatype.html for more detail.
+// Publish to Bintray
+ThisBuild / description := "Glow: Genomics on Apache Spark"
+ThisBuild / homepage := Some(url("http://projectglow.io"))
```
Comment (henrydavidge) [inline build.sbt]: I think I fixed the https thing

File: build.sbt [hunk 5]
```diff
@@ -127,33 +158,37 @@ lazy val python =
...
         require(ret == 0, "Python tests failed")
-      }
+      },
+      publish / skip := true
     )
 
-// Uncomment the following for publishing to Sonatype.
-// See https://www.scala-sbt.org/1.x/docs/Using-Sonatype.html for more detail.
+// Publish to Bintray
+ThisBuild / description := "Glow: Genomics on Apache Spark"
```
Comment (henrydavidge) [inline build.sbt]: Fix description here too please

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,10 @@
+==================================
+Glow: Genomics on Apache Spark SQL
```
Comment (henrydavidge) [inline README.rst]: Fix description here

File: README.md [hunk 7]
```diff
@@ -1,36 +0,0 @@
-[![CircleCI](https://circleci.com/gh/projectglow/glow.svg?style=svg&circle-token=7511f70b2c810a18e88b5c537b0410e82db8617d)](https://circleci.com/gh/projectglow/glow)
-
-# Building and Testing
```
Comment (henrydavidge) [inline README.md]: Ok, I think I was incorrect before when I said that we should get rid of this. This is important information that people need to contribute to the project. I guess the Pypi readme is more about usage.

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+==================================
+
+|circle-ci|
+
+.. |circle-ci| image:: https://circleci.com/gh/projectglow/glow.svg?style=svg&circle-token=7511f70b2c810a18e88b5c537b0410e82db8617d
+    :target: https://circleci.com/gh/projectglow/glow
+
+Glow is an open-source toolkit for large-scale genomic analysis.
```
Comment (henrydavidge) [inline README.rst]: Link to the documentation?

File: python/setup.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+from setuptools import setup
+
+setup(
+    name='glowpy',
```
Comment (henrydavidge) [inline python/setup.py]: nit: can this be `glow.py` or `glow-py`? `glowpy` looks kind of ugly to me

File: core/src/test/resources/log4j.properties [hunk 10]
```diff
@@ -9,3 +9,4 @@ log4j.appender.file.File=unit-tests.log
 log4j.logger.akka=ERROR
 log4j.logger.Remoting=ERROR
 log4j.logger.org.eclipse.jetty=ERROR
+log4j.logger.org.apache.hadoop=ERROR
```
Comment (henrydavidge) [inline core/src/test/resources/log4j.properties]: `WARN`? I wonder if there isn't something valuable here.

File: README.rst [hunk 11]
```diff
@@ -0,0 +1,10 @@
+==================================
```
Comment (henrydavidge) [inline README.rst]: Can you move this into `python` to make clear that it's the pypi readme?

File: version.sbt (hunk 1 above)
Comment (karenfeng) [reply version.sbt]: This is part of the `sbt release` process.

**Pull request merged**